src/pages/ResultsPage.js

```diff
@@ -188,6 +146,73 @@ class BaseResultsPage extends BaseCategoriesPage {
             }
         }
     }
+
+    onBackClick(event: SyntheticInputEvent): void {
+        if (this.refs.component.onGoBack) {
+            this.refs.component.onGoBack(event)
+        }
+        if (!event.defaultPrevented) {
+            this.props.history.goBack();
+        }
+    }
+
+    render(): ReactElement {
+        const Component = this
+            .props
+            .location
+            .pathname
+            .match(/map(\/)?$/) ? ResultsMap
+            : ResultsListPage;
```

Comment:
Your other option is to make the list and map views nested in the main page view and then include another level of `<RoutingContext>`. You can then instead of sharing the object in props, share it in context.